apps/feelslike/feels_like.star

```diff
@@ -0,0 +1,118 @@
+"""
+Applet: Feels Like
+Summary: Shows "feels like" weather
+Description: Displays the "feels like" temperature for the set location.
+Author: skalum
+"""
+
+load("cache.star", "cache")
+load("encoding/json.star", "json")
+load("http.star", "http")
+load("render.star", "render")
+load("schema.star", "schema")
+
+DEFAULT_LOCATION = """
+{
+	"lat": "37.7639316",
+	"lng": "-122.4030647",
+	"description": "451 Kansas St, San Francisco, CA 94107, USA",
+	"locality": "San Francisco",
+	"place_id": "ChIJSbhWnuF_j4ARbk1kioTNmEE",
+	"timezone": "America/Los_Angeles"
+}
+"""
+
+DEFAULT_API_KEY = "noKey"
+
+DEFAULT_TTL_SECONDS = "7200"
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "apiKey",
+                name = "API key",
+                desc = "OpenWeather API key",
+                icon = "key",
+            ),
+            schema.Dropdown(
+                id = "updateInterval",
+                name = "Update interval",
+                desc = "Interval at which to update weather information",
+                icon = "clock",
+                default = "7200",
+                options = [
+                    schema.Option(
+                        display = "Two hours",
+                        value = "7200",
+                    ),
+                    schema.Option(
+                        display = "One hour",
+                        value = "3600",
+                    ),
+                    schema.Option(
+                        display = "Ten minutes",
+                        value = "600",
+                    ),
+                ],
+            ),
+            schema.Location(
+                id = "location",
+                name = "Location",
+                desc = "Location for which to display weather",
+                icon = "locationDot",
+            ),
+        ],
+    )
+
+def main(config):
+    currentWeather = cache.get("currentWeather")
```

Comment:
Bit of a problem here I'm afraid. The cache is shared across all "installations"of the app, so this cache record will be loaded and displayed by everyone running the app. The way around it is to include something in the key that's unique to the data stored, e.g. latitude and longitude.

Reply:
Wanted to mention that we recently added caching to the http client itself. I think that'll greatly simplify the cache handling in this app. You can read more about it [over here](https://discuss.tidbyt.com/t/http-caching-client/5129).